community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/mutation/MergePatternAction.scala

```diff
@@ -120,16 +121,16 @@ case class MergePatternAction(patterns: Seq[Pattern],
     Effects(collect.toSet)
   }
 
-
   def localEffects(externalSymbols: SymbolTable) = {
     import Effects._
 
-    val allSymbols = updateSymbols(externalSymbols)
+    val effectsFromReading = readEffects(externalSymbols)
 
+    val allSymbols = updateSymbols(externalSymbols)
     val actionEffects = actions.effects(allSymbols)
     val onMatchEffects = onMatch.effects(allSymbols)
     val updateActionsEffects = updateActions.effects(allSymbols)
-    val effectsFromReading = readEffects(allSymbols)
+
```

Comment:
It's not just a reorder, the argument to the `readEffects` function has also changed, from `allSymbols` to `externalSymbols`. We reordered it to clarify that it does not work on the same `SymbolTable` as the other effect calculation functions.